Tasks/VsTest/vstest.ts

```diff
@@ -797,60 +503,53 @@ function invokeVSTest(testResultsDirectory: string): Q.Promise<number> {
-            then(function (ret) {
-                newSettingsFile = ret;
-                runVStest(testResultsDirectory, newSettingsFile, vsVersion)
-                    .then(function (code) {
-                        if (code !== 0) {
-                            utils.Helper.publishEventToCi(AreaCodes.INVOKEVSTEST, 'RunVStest returned ' + code, 1034, true);
-                        }
-                        defer.resolve(code);
-                    })
-                    .fail(function (code) {
-                        if (code !== 0) {
-                            utils.Helper.publishEventToCi(AreaCodes.INVOKEVSTEST, 'RunVStest returned ' + code, 1035, false);
-                        }
-                        defer.resolve(code);
-                    });
-            });
-    } catch (error) {
-        tl.warning(tl.loc('ErrorWhileUpdatingSettings'));
-        tl.debug(error);
+        newSettingsFile = settingsHelper.updateSettingsFileAsRequired(vstestConfig.settingsFile, vstestConfig.runInParallel, vstestConfig.tiaConfig, vstestConfig.vsTestVersionDetails, false, vstestConfig.overrideTestrunParameters, false);
+        return vsTestCall(newSettingsFile, vsVersion);
+    }
+    catch (err) {
         //Should continue to run without the selected configurations.
-        runVStest(testResultsDirectory, newSettingsFile, vsVersion)
-            .then(function (code) {
-                if (code !== 0) {
-                    utils.Helper.publishEventToCi(AreaCodes.INVOKEVSTEST, 'RunVStest returned ' + code, 1036, false);
-                }
-                defer.resolve(code);
-            })
-            .fail(function (code) {
-                if (code !== 0) {
-                    utils.Helper.publishEventToCi(AreaCodes.INVOKEVSTEST, 'RunVStest returned ' + code, 1037, false);
-                }
-                defer.resolve(code);
-            });
+        return vsTestCall(newSettingsFile, vsVersion);
     }
+}
 
-    return defer.promise;
+function vsTestCall(newSettingsFile, vsVersion): Q.Promise<tl.TaskResult> {
+    return runVStest(newSettingsFile, vsVersion).then(function (code) {
+        if (code !== 0) {
+            utils.Helper.publishEventToCi(AreaCodes.INVOKEVSTEST, 'RunVStest returned ' + code, 1036, false);
+            return tl.TaskResult.Failed;
+        }
+        return tl.TaskResult.Succeeded;
+    }).catch(function (err) {
+        utils.Helper.publishEventToCi(AreaCodes.INVOKEVSTEST, err.message, 1037, false);
+        tl.error(err);
+        return tl.TaskResult.Failed;
+    });
 }
 
-function publishTestResults(testResultsDirectory: string) {
-    if (testResultsDirectory) {
-        const resultFiles = tl.findMatch(testResultsDirectory, path.join(testResultsDirectory, '*.trx'));
+function publishTestResults(testResultsDirectory: string): tl.TaskResult {
```

Comment:
Are we making sure that the results directory used is the updated (re-calculated) one. Example scenarios to validate: Use a runsettings with test results location as a relative path. Turn on different settings options such as TIA , run in parallel

Reply:
Yes will verify this. Good bash scenario